game/engine.py

```
@@ -295,18 +299,60 @@ def _call_engine(self, command):
+                try:
+                    if proc:
+                        proc.kill()
+                except Exception:
+                    logger.exception('Failed to kill timed-out engine process')
+                try:
+                    if proc:
+                        out, err = proc.communicate(timeout=1)
+                        logger.debug('Partial stdout after timeout: %s', (out or '').strip())
+                        logger.debug('Partial stderr after timeout: %s', (err or '').strip())
+                except Exception:
+                    pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)